check-http/lib/check_http.go

```diff
@@ -105,6 +107,14 @@ func run(args []string) *checkers.Checker {
 		},
 		Proxy: http.ProxyFromEnvironment,
 	}
+	if opts.SourceIP != "" {
+		tr.Dial = (&net.Dialer{
+			LocalAddr: &net.TCPAddr{IP: net.ParseIP(opts.SourceIP)},
```

Comment:
`net.ParseIP` returns nil if the given string is not a valid representation of an IP address. In such cases, returning `Unknown` would be plausible in my opinion.